lib/scan/detect_values.rb

```diff
@@ -23,6 +23,22 @@ def self.set_additional_default_values
       return config
     end
 
+    def self.filter_simulators(simulators, deployment_target)
+      # Filter out any simulators that are not the same major and minor version of our deployment target
+      if deployment_target.to_s.length > 0
+        components = deployment_target.split('.')
+        major = (components.length >= 1) ? components[0] : 0
+        minor = (components.length >= 2) ? components[1] : 0
+        simulators = simulators.select do |s|
+          sim_components = s.ios_version.split('.')
+          sim_major = (sim_components.length >= 1) ? sim_components[0] : 0
+          sim_minor = (sim_components.length >= 2) ? sim_components[1] : 0
+          (sim_major > major) || ((sim_minor >= minor) && (sim_major == major))
+        end
```

Comment:
I think much of this has the potential to be doing string comparison for these version pieces, which seems bad.

If we make use of the `Gem::Version` class, can we the needed comparison logic for free?

```
irb(main):012:0> v1 = Gem::Version.new("9.2")
=> #<Gem::Version "9.2">
irb(main):013:0> v2 = Gem::Version.new("9.0")
=> #<Gem::Version "9.0">
irb(main):014:0> v1 > v2
=> true
irb(main):019:0> v4 = Gem::Version.new("10.0")
=> #<Gem::Version "10.0">
irb(main):020:0> v1 > v4
=> false
```